packages/js-runtime/typescript/compiler.ts

```diff
@@ -300,8 +303,36 @@ export class TypeScriptCompiler implements Compiler<TypeScriptCompilerOptions> {
       throw new Error("Missing main source.");
     }
 
+    // Check if the main source file has the /// <cts-enable /> directive
+    const hasCtsEnableDirective = (sourceFile: SourceFile): boolean => {
```

Comment:
This doesn't need to be defined dynamically inside the enclosing function/no enclosed variables, `compile` function is long enough, can move to bottom of file or elsewhere